mwclient/page.py

```diff
@@ -204,9 +205,21 @@ def save(self, text, summary=u'', minor=False, bot=True, section=None, **kwargs)
             data['assert'] = 'user'
 
         def do_edit():
-            result = self.site.post('edit', title=self.name, text=text,
-                                    summary=summary, token=self.get_token('edit'),
-                                    **data)
+            result = None
+            if edit_type == 'replace':
+                result = self.site.post('edit', title=self.name, text=text,
```

Comment:
Hm, it's not so elegant to repeat this code three times. I wonder if it would be better to add two new methods `append(text)` and `prepend(text)` that calls `save(...)`

Reply:
How about this:
```
    def save(self, text, summary=u'', minor=False, bot=True, section=None, **kwargs):
        """Update the text of a section or the whole page by performing an edit operation.
                """
        return self._save(summary, minor, bot, section, text=text, **kwargs)
    
    def save_append(self, text, summary=u'', minor=False, bot=True, section=None,
                    **kwargs):
        """Append text to the text of a section or the whole page by performing an edit operation.
                """
        return self._save(summary, minor, bot, section, append_text=text, **kwargs)
    
    def save_prepend(self, text, summary=u'', minor=False, bot=True, section=None,
                     **kwargs):
        """Prepend text to the text of a section or the whole page by performing an edit operation.
                """
        return self._save(summary, minor, bot, section, prepend_text=text, **kwargs)
```

and then below in _save,
```
 def do_edit():
            result = self.site.post('edit', title=self.name, summary=summary,
                                    token=self.get_token('edit'),
                                        **data)
```

I didn't test yet but if that looks good to you I'll run tests, fix the docstring for touch, and send a new commit - sorry for the whitespace issues, I'll try to fix those too.

Reply:
Did you see my other suggestion above? I think I would prefer having methods named just `append` and `prepend` rather  than `save_prepend` and `save_append`. Do you agree?

Reply:
Prepend/Append didn't sound obviously enough like page-saving actions to me out of context, and if they start with save_ then they'd show up in autocomplete along with save, is why I renamed them. But if you'd still prefer just prepend/append I'm fine with that just to have simpler names for everything.

Reply:
I still find the `save_` names a bit clumsy, but it sounds useful to have them show up in autocomplete along with save, yes, I agree. We could try getting a third opinion on this. @waldyrious, do you have a preference here?

Reply:
I understand the motivation, but I am not sure the `save_` prefix is the best solution to the problem, for a few  reasons:

1. We already have other page-changing methods that aren't prefixed with `save_`, such as `delete()` and `move()`
2. "save" is actually a little ambiguous, since it doesn't imply that the text is changed. It sounds more like a method that's part of a framework that supported piling up local changes and then a `save()` method to commit them.
3. The names `save_prepend` and `save_append` are, indeed, a little odd-sounding IMO. They don't feel like part of a well-planned API.

My suggestion would be to follow-up #51 and rename `save()` to `edit()` (which should be reasonable, since it's been three releases since the one that changed `edit()` to `text()`), and keep the two new methods named `prepend()` and `append()`.

How does that sound?

Reply:
Would you keep save() as an alias to edit() for backwards compatibility? if so that sounds good, otherwise I'm not sure it's worth breaking compatibility to rename save, save -> edit is only a small improvement in clarity compared to edit -> text.

Reply:
Sure, no objections to keeping `save()` as an alias for compatibility.

Reply:
Ok great, I have a pretty bad cold right now but I'll send new changes by the end of the week:
* edit, prepend, and append will call _edit
* _edit will be similar to current save except text/append_text/prepend_text will be part of **kwargs
* alias of save for edit
* update touch to work with new method & add clarity in docstring

Reply:
Wish you a quick recovery  :sparkles:

Sounds like a good plan. My first thought was that `append` and `prepend` could just call <s>`save`</s> `edit(text=None, ...)`, but I'm open to the idea of adding a new `_edit` method if you think that's less messy :)

Reply:
Ok made the changes. Do you have a settings file I can import to Pycharm so I have the right whitespace settings for everything?

Reply:
> Do you have a settings file I can import to Pycharm so I have the right whitespace settings for everything?

`.editorconfig` files are [natively supported by PyCharm](https://editorconfig.org/#download), so it should have picked up the definitions [we have already](https://github.com/mwclient/mwclient/blob/master/.editorconfig). Maybe there are additional configurations we need to add to the file?